src/seedu/addressbook/Main.java

```diff
@@ -1,160 +1,140 @@
+             */
+            throw new RuntimeException(e);
         }
     }
 
     /**
-     * Entry point.
+     * Prints the Goodbye message and exits.
      */
-    public static void main(String... launchArgs) {
-        System.out.println(MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE);
-        try {
-            final Main main = new Main(getStorageFilePathFromLaunchArgs(launchArgs), System.in, System.out);
-            main.start();
-        } catch (MainInitialisationException mie) {
-            System.out.println(MESSAGE_INIT_FAILED);
-        }
+    private void exit() {
+        ui.showGoodbyeMessage();
+        System.exit(0);
     }
 
     /**
-     * Retrieves the string representing the intended storage file path as specified in the launch args.
-     * Defaults to {@link #DEFAULT_STORAGE_FILEPATH} if no storage file argument is found.
-     * 
-     * @param launchArgs full program launch arguments passed to application main method
+     * Reads the user command and executes it, until the user issues the exit command.
      */
-    private static String getStorageFilePathFromLaunchArgs(String... launchArgs) {
-        return launchArgs.length > 0 ? launchArgs[0] : DEFAULT_STORAGE_FILEPATH;
-    }
+    private void runCommandLoopUntilExitCommand() {
+        Command command;
+        do {
+            String userCommand = ui.getUserCommand();
+            command = new Parser().parseCommand(userCommand);
+            CommandResult result = executeCommand(command);
+            updateLastShownList(result);
+            ui.showResultToUser(result);
 
-    /**
-     * Displays the goodbye message and exits the runtime.
-     */
-    private void exitProgram() {
-        Command exit = new ExitCommand();
-        exit.injectDependencies(ui, addressBook);
-        exit.execute();
+        } while(!ExitCommand.isExit(command));
     }
 
-    /*
-     * ==============NOTE TO STUDENTS======================================
-     * Notice how this method solves the whole problem at a very high level.
-     * We can understand the high-level logic of the program by reading this
-     * method alone.
-     * ====================================================================
-     */
-
     /**
-     * Starts program execution after all dependencies and components successfully initialised.
+     * Updates the {@link #lastShownList} if the result contains a list of Persons
      */
-    public void start() {
-        ui.showWelcomeMessage(VERSION);
-        ui.showToUser(String.format(MESSAGE_USING_STORAGE_FILE, storageFile.toString()));
-        // [read input, execute, response] loop
-        while (true) {
-            String userCommand = ui.getUserCommand();
-            ui.echoLastEnteredUserCommand();
-            String feedback = executeCommand(userCommand);
-            ui.showResultToUser(feedback);
+    private void updateLastShownList(CommandResult result) {
+        if(result.getRelevantPersons() != null) {
```

Comment:
Yes, this is a good place for `Optional`, but let's not introduce it in this version. That kind of 'optional' topics can be delayed. In early levels we should limit ourselves to essentials only.